numba/np/npyfuncs.py

```diff
@@ -753,6 +768,25 @@ def np_complex_square_impl(context, builder, sig, args):
                                      [args[0], args[0]])
 
 
+########################################################################
+# NumPy cbrt
+
+def np_real_cbrt_impl(context, builder, sig, args):
+    import numpy as np
+
+    _check_arity_and_homogeneity(sig, args, 1)
+
+    def float_cbrt(x):
+        if np.isnan(x):
+            return x
+        if x < 0:
+            return -np.power(-x, 1.0 / 3.0)
+        else:
+            return np.power(x, 1.0 / 3.0)
```

Comment:
This is going to be "interesting" in that in NumPy, if a system has a `cbrt` from C99 then it gets used, else something like what you've written is used. The impact being that e.g. signed zero is incorrect for me locally as I've got C99 `cbrt`!
```python
from numba import njit
import numpy as np

@njit
def foo():
    return np.cbrt(-0.)

print(foo.py_func())
print(foo())
```
gives:
```
-0.0
0.0
```

LLVM can be coerced into generating `cbrt` with appropriate hints.

Reply:
> LLVM can be coerced into generating `cbrt` with appropriate hints.

`np.signbit` can be used here to distinguish `0.0` from its negative counterpart. I didn't want to use it since the numba implementation uses the [numpy one](https://github.com/numpy/numpy/blob/master/numpy/core/src/npymath/_signbit.c) and I'd like to make this function available in RBC.  

One could mimic `npy_signbit` [in pure LLVM](https://godbolt.org/z/nd7hj9) but that would require knowledge about the system (`sizeof(int)` and endianess).

Reply:
Replaced comparison by `np.signbit` in my last commit. I think this should fix signed zero comparison.

Reply:
The issues I see with this are:

* LLVM will generate a call to `cbrt` under certain conditions and NumPy is using `libm`'s `cbrt` if it's available, so this would match exactly. Generating the call correctly would also permit e.g. use of SVML's `cbrt`. If you need a hand working out how to do this then shout! I'd recommend looking at the impact of `fastmath` flags as a starting point.
* If RBC needs this then it should implement it or make Numba generic to the point that RBC can consume it?

Reply:
> If RBC needs this then it should implement it or make Numba generic to the point that RBC can consume it?

I think it is consumable in the way it is today. We already ship our own version of `np.signbit`. By the way, I just remember that the libm version of cbrt works on RBC. So, no reason for not using it.

> LLVM will generate a call to cbrt under certain conditions and NumPy is using libm's cbrt if it's available, so this would match exactly. Generating the call correctly would also permit e.g. use of SVML's cbrt. If you need a hand working out how to do this then shout! I'd recommend looking at the impact of fastmath flags as a starting point.

I will replace the code by a call to libm cbrt.

Reply:
@stuartarchibald, I notice that a few functions calls the numpy implementation (i.e. `npy_logaddexp`). Is there any reason for not calling libm directly?

Reply:
> > LLVM will generate a call to cbrt under certain conditions and NumPy is using libm's cbrt if it's available, so this would match exactly. Generating the call correctly would also permit e.g. use of SVML's cbrt. If you need a hand working out how to do this then shout! I'd recommend looking at the impact of fastmath flags as a starting point.
> 
> I will replace the code by a call to libm cbrt.

What guarantees that `cbrt` is in a `libm` ?

Reply:
Isn't `cbrt` part of `libm`?

Reply:
If you have C99. NumPy does this:
https://github.com/numpy/numpy/blob/31ffdecf07d18ed4dbb66b171cb0f998d4b190fa/numpy/core/src/npymath/npy_math_internal.h.src#L508-L529

Reply:
Makes sense! I will update the PR

Reply:
Do you have any clue of why CI is failing with `npy_cbrt`?

```C
Thread 1 "python" received signal SIGSEGV, Segmentation fault.
0x0000000000000000 in ?? ()
(gdb) bt
#0  0x0000000000000000 in ?? ()
#1  0x00007ffff7dfc051 in cpython::__main__::foo$241(long long) ()
#2  0x00007ffff3b1cb3c in call_cfunc (self=0x7fffeb670c10, cfunc=0x7fffd24648b0, args=0x7fffd24c2f10, kws=0x0, locals=0x0) at numba/_dispatcher.c:353
#3  0x00007ffff3b1d6f5 in Dispatcher_call (self=0x7fffeb670c10, args=0x7fffd24c2f10, kws=<optimized out>) at numba/_dispatcher.c:576
#4  0x000055555568395f in _PyObject_MakeTpCall () at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Objects/call.c:159
#5  0x0000555555735300 in _PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x7ffff7eeb5b8, callable=0x7fffeb670c10)
    at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Include/cpython/abstract.h:125
#6  call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x5555558efa90)
    at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Python/ceval.c:4963
#7  _PyEval_EvalFrameDefault () at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Python/ceval.c:3500
#8  0x000055555571c190 in PyEval_EvalFrameEx (throwflag=0, f=0x7ffff7eeb440)
    at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Python/ceval.c:741
#9  _PyEval_EvalCodeWithName () at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Python/ceval.c:4298
#10 0x000055555571d9f3 in PyEval_EvalCodeEx (closure=0x0, kwdefs=0x0, defcount=0, defs=0x0, kwcount=0, kws=0x0, argcount=0, args=0x0,
    locals=<optimized out>, globals=<optimized out>, _co=<optimized out>)
    at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Python/ceval.c:4327
#11 PyEval_EvalCode (co=<optimized out>, globals=<optimized out>, locals=<optimized out>)
    at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Python/ceval.c:718
#12 0x00005555557908b2 in run_eval_code_obj () at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Python/pythonrun.c:1125
#13 0x00005555557a3492 in run_mod () at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Python/pythonrun.c:1147
#14 0x00005555557a657e in PyRun_FileExFlags () at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Python/pythonrun.c:1063
#15 0x00005555557a6769 in PyRun_SimpleFileExFlags () at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Python/pythonrun.c:428
#16 0x00005555557a6c1e in pymain_run_file (cf=0x7fffffffc0c8, config=0x5555558eec20)
    at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Modules/main.c:387
#17 pymain_run_python (exitcode=0x7fffffffc0c0) at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Modules/main.c:612
#18 Py_RunMain () at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Modules/main.c:691
#19 0x00005555557a6e19 in Py_BytesMain () at /home/conda/feedstock_root/build_artifacts/python_1596159888721/work/Modules/main.c:1123
#20 0x00007ffff77e6b97 in __libc_start_main (main=0x55555565a7f0 <main>, argc=2, argv=0x7fffffffc2b8, init=<optimized out>, fini=<optimized out>,
    rtld_fini=<optimized out>, stack_end=0x7fffffffc2a8) at ../csu/libc-start.c:310
#21 0x000055555574aba9 in _start ()
(gdb)
```